specifyweb/frontend/js_src/lib/components/Forms/Save.tsx

```diff
@@ -214,27 +219,81 @@ export function SaveButton<SCHEMA extends AnySchema = AnySchema>({
     </ButtonComponent>
   );
 
+  const [carryForwardAmount, setCarryForwardAmount] = React.useState<number>(2);
+  const [showBulkCarryInput, setBulkCarryInput] = React.useState(false);
+
   return (
     <>
       {typeof handleAdd === 'function' && canCreate ? (
         <>
+          {resource.specifyTable.name === 'CollectionObject' &&
+          (isInRecordSet === false || isInRecordSet === undefined) &&
+          showBulkCarryInput ? (
+            <Label.Inline>
+              <Input.Generic
+                autoFocus
+                forwardRef={(input): void => input?.focus()}
+                id={id('bulk-amount')}
+                type="number"
+                value={carryForwardAmount}
+                onValueChange={(value): void =>
+                  carryForwardAmount === undefined
+                    ? setCarryForwardAmount(2)
+                    : setCarryForwardAmount(Number.parseInt(value))
+                }
+              />
+            </Label.Inline>
+          ) : null}
+          {resource.specifyTable.name === 'CollectionObject' &&
+          (isInRecordSet === false || isInRecordSet === undefined) &&
+          !resource.needsSaved ? (
+            <Button.Secondary
+              aria-controls={id('bulk-amount')}
+              title={commonText.bulkCarry()}
+              onClick={(): void => setBulkCarryInput(!showBulkCarryInput)}
+            >
+              {commonText.bulkCarry()}
+            </Button.Secondary>
+          ) : null}
           {showClone &&
             copyButton(
               formsText.clone(),
               formsText.cloneDescription(),
-              async () => resource.clone(true)
+              async () => [await resource.clone(true)]
             )}
           {showCarry &&
             copyButton(
               formsText.carryForward(),
               formsText.carryForwardDescription(),
-              async () => resource.clone(false)
+              resource.specifyTable.name === 'CollectionObject' &&
+                carryForwardAmount > 2
+                ? async () => {
+                    const formatter =
+                      tables.CollectionObject.strictGetLiteralField(
+                        'catalogNumber'
+                      ).getUiFormatter()!;
+                    const wildCard = formatter.valueOrWild();
```

Comment:
Hmm should this feature only be allowed for auto-incrementing catalog numbers?

Reply:
yes